From 8310482e5c46d6a15894c4d3ebf8156264175282 Mon Sep 17 00:00:00 2001
From: Ingo Franzki <ifranzki@linux.ibm.com>
Date: Wed, 12 Oct 2022 09:00:51 +0200
Subject: [PATCH 02/34] EP11: Do not report DSA/DH parameter generation as
 being supported

Mechanisms CKM_DSA_PARAMETER_GEN and CKM_DH_PKCS_PARAMETER_GEN were
reported to be supported in the mechanism list of the EP11 token, but
they never were really supported in the code.

Remove them from the mechanism list.

Signed-off-by: Ingo Franzki <ifranzki@linux.ibm.com>
---
 usr/lib/ep11_stdll/ep11_specific.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/usr/lib/ep11_stdll/ep11_specific.c b/usr/lib/ep11_stdll/ep11_specific.c
index 5d7c5607..737b373b 100644
--- a/usr/lib/ep11_stdll/ep11_specific.c
+++ b/usr/lib/ep11_stdll/ep11_specific.c
@@ -6669,7 +6669,6 @@ CK_RV ep11tok_generate_key_pair(STDLL_TokData_t * tokdata, SESSION * sess,
                                      public_key_obj->template,
                                      private_key_obj->template);
         break;
-    case CKM_DSA_PARAMETER_GEN:
     case CKM_DSA_KEY_PAIR_GEN:
         rc = dsa_generate_keypair(tokdata, sess, pMechanism,
                                   public_key_obj->template,
@@ -9042,10 +9041,8 @@ static const CK_MECHANISM_TYPE ep11_supported_mech_list[] = {
     CKM_DES3_KEY_GEN,
     CKM_DH_PKCS_DERIVE,
     CKM_DH_PKCS_KEY_PAIR_GEN,
-    CKM_DH_PKCS_PARAMETER_GEN,
     CKM_DSA,
     CKM_DSA_KEY_PAIR_GEN,
-    CKM_DSA_PARAMETER_GEN,
     CKM_DSA_SHA1,
     CKM_EC_KEY_PAIR_GEN,
     CKM_ECDH1_DERIVE,
-- 
2.16.2.windows.1

